Dolby/ott

ShaoWeiguo opened the pull request:
Support Dolby Codecs in Bento4

muxer: adding support DD, DD+, AC-4 and Dolby Vision
fragment, mp4info, mp42hls etc: fix the issues for Dolby codecs
manifest: support Dolby codecs in DASH and HLS manifest

krzemienski commented:
this is awesome work @ShaoWeiguo ...if there is anything i can do to help this the work please let me know! I was going to try and work on `manifest: support Dolby codecs in DASH and HLS manifest` but seems like you have handled.

I have a decent amount of encoding ladder artifacts that were encoded by x265 and x265 that range from pq->sdr and also from 4k down that I can run this branch against if that is any help!

barbibulle commented:
Thanks for the PR. I will most likely merge the new features in separate discrete releases so that this can be tested incrementally.

ShaoWeiguo commented:
@krzemienski  Sure, thanks.

ShaoWeiguo commented:
Dolby made some changes about Dolby Vision signaling in MP4. So please change the codes below in Mp4Mux.cpp.

- line 1218, const AP4_UI32 dv_major_version = 2; -- > const AP4_UI32 dv_major_version = 1;
- line 1219, const AP4_UI32 dv_minor_version = 1; --> const AP4_UI32 dv_minor_version = 0;
- line 1757, const AP4_UI32 dv_major_version = 2; -- > const AP4_UI32 dv_major_version = 1;
- line 1758, const AP4_UI32 dv_minor_version = 1; --> const AP4_UI32 dv_minor_version = 0;

barbibulle commented:
@ShaoWeiguo The PR was made before the code in the repo was updated to python3. Would it be possible to rebase and update your PR so that the conflicts can be resolved?

ShaoWeiguo commented:
I see. I will update the PR and make it base on latest commit. Need several days effort, and wish there is no **big changes** in master branch during this period.

ShaoWeiguo commented:
@barbibulle  I have updated the PR based on latest commit (17e26f74fae859098faf7bcac45c3cb84d4bb8e7). And I also fixed the Dolby Vision version issue mentioned above. If anything need, please let me know.

barbibulle commented:
> @barbibulle I have updated the PR based on latest commit ([17e26f7](https://github.com/axiomatic-systems/Bento4/commit/17e26f74fae859098faf7bcac45c3cb84d4bb8e7)). And I also fixed the Dolby Vision version issue mentioned above. If anything need, please let me know.

Thanks @ShaoWeiguo, I'll work on this ASAP.

jimmymic commented:
@barbibulle Is there anything we can do to assist getting this PR completed?

barbibulle commented:
@jimmymic I have started merging the changing in an integration branch. It is taking me some time because the PR is pretty large, so I have to check that merging it won't break something else. I'm 50% done, and will keep working on it next week, so I would expect something soon.

ShaoWeiguo commented:
@barbibulle Any update？

barbibulle commented:
Hi. Sorry about the delay. Most the merging is done but I have been delayed on finishing it. I will attend to it ASAP and let you know when the merged code is in the master branch.

ShaoWeiguo commented:
finish the merge pull request?

barbibulle commented:
Hi. I have merged all the changes to the C++ code. I have not yet been able to merge the python changes, because that's more difficult. The changes don't match the code style/architecture directly, so I'll likely will have to rewrite something equivalent that doesn't deviate so much. The issue I'm facing is that there's no test suite for the Dolby features, so I don't know how to test that my version would be equivalent to what's in the PR. Would it be possible to come up with a few files and scripts to have at least some basic checks/tests?
